core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -40,7 +40,7 @@
 
   public SqlNodeList keywordList;
   public SqlNode topN;
-  SqlNode exceptExpression;
+  public final SqlNode exceptExpression;
```

Comment:
nit: Let's keep the fields in the same order as the constructor args (e.g. exceptExpression after selectList).